### PR TITLE
fix the memory leak in handler.streamingReq and handler.streamingRes

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal 
#1060

### CODE REVIEW

File: node/v2/handler.js [hunk 1]
```diff
@@ -325,9 +331,15 @@ TChannelV2Handler.prototype.handleCallResponse = function handleCallResponse(res
     if (handled) {
         if (res.state === States.Streaming) {
             self.streamingRes[res.id] = res;
+            res.finishEvent.on(onResponseDone);
+            res.errorEvent.on(onResponseDone);
         }
```
Comment (Raynos) [inline node/v2/handler.js]: You could just do `else { delete self.streamingRes[id]; }`

**Comment by kriskowal** [comment]
:ok: Curious about @raynos’s observation. This approach seems intuitive.

File: node/v2/handler.js (hunk 1 above)
Comment (ShanniLi) [inline node/v2/handler.js]: if we do else branch, then we need to do it both on if (handled)  and if (res.state === States.Streaming).

And as @kriskowal pointed out, the "else" approach is less intuitive.

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**